src/internal/Util.ts

```diff
@@ -69,4 +69,15 @@ export class Util {
       return undefined;
     }
   }
+
+  /**
+   * Encodes the parameters.
+   * @param pramaters the parameters to encode
+   */
+  public static encodeParameters(pramaters: any) {
+      Object.keys(pramaters).forEach((key, index) => {
+          pramaters[key] = encodeURIComponent(pramaters[key]);
```

Comment:
`encodeURIComponent` is the right way to do this, yeah.  I did some digging for the Axios side and it does appear that they don't do it for you, so it should be correct to encode it for all implementations.